oracle/oracle.go

```diff
@@ -490,6 +541,79 @@ func (o *Oracle) checkAcceptList(params oracletypes.Params) {
 	}
 }
 
+func (o *Oracle) checkCurrencyPairAndDeviations(currentParams, newParams oracletypes.Params) (err error) {
+	if currentParams.CurrencyPairProviders.String() != newParams.CurrencyPairProviders.String() {
+		o.logger.Debug().Msg("Updating Currency Pair Providers Map")
+		o.providerPairs = createPairProvidersFromCurrencyPairProvidersList(newParams.CurrencyPairProviders)
+	}
+	if currentParams.CurrencyDeviationThresholds.String() != newParams.CurrencyDeviationThresholds.String() {
+		o.logger.Debug().Msg("Updating Currency Deviation Thresholds Map")
+		o.deviations, err = createDeviationsFromCurrencyDeviationThresholdList(newParams.CurrencyDeviationThresholds)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// createPairProvidersFromCurrencyPairProvidersList will create the pair providers
+// map used by the price feeder Oracle from a CurrencyPairProvidersList defined by
+// Ojo's oracle module.
+func createPairProvidersFromCurrencyPairProvidersList(
```

Comment:
nit: This oracle file is pretty huge and these two functions are not methods of the Oracle struct.I'd prefer them in their own file or another one with related functions.

Reply:
Moved to utils👍